addons/l10n_in/models/account_invoice.py

```diff
@@ -50,17 +50,22 @@ def _compute_l10n_in_gst_treatment(self):
                 record.l10n_in_gst_treatment = gst_treatment
         (self - indian_invoice).l10n_in_gst_treatment = False
 
-    @api.depends('partner_id', 'company_id')
+    @api.depends('partner_id', 'partner_shipping_id', 'company_id')
     def _compute_l10n_in_state_id(self):
         for move in self:
             if move.country_code == 'IN' and move.journal_id.type == 'sale':
-                country_code = move.partner_id.country_id.code
+                partner_state = (
+                    move.partner_id.commercial_partner_id == move.partner_shipping_id.commercial_partner_id
+                    and move.partner_shipping_id.state_id
+                    or move.partner_id.state_id
+                )
+                if not partner_state:
+                    partner_state = move.partner_id.commercial_partner_id.state_id or move.company_id.state_id
+                country_code = partner_state.country_id.code
                 if country_code == 'IN':
-                    move.l10n_in_state_id = move.partner_id.state_id
-                elif country_code:
-                    move.l10n_in_state_id = self.env.ref('l10n_in.state_in_oc', raise_if_not_found=False)
+                    move.l10n_in_state_id = partner_state
```

Comment:
@hamo-odoo Is not it more logical that if there is no country set, we take the state from the company as before?  Why do we change that logic?

Reply:
@jco-odoo on L63 we already handled the fallback for that in the worst case we are considering the `company_id` isn't it obvious country_code will be `IN` in that case?

Reply:
But what if you have the state set, but not the country, the state will change to overseas.  (although you have to do an effort to have a state without country filled in)

Reply:
And what if the country and state are not set, then it will also fallback to overseas instead of company state.